2-scrape-metadata/index.js

```diff
@@ -27,15 +28,15 @@ var pageData = {
 };
 
 converter.fromFile( CSV_FILENAME, function( err, result ) {
-    var eisIdArray = result.map( function( obj ) {
-	return obj.eis_id;
+  var eisIdArray = result.map( function( obj ) {
+    return obj.eis_id;
   } );
-
-    eisIdArray.forEach( function( id ) {
-	request( 'https://cdxnodengn.epa.gov/cdx-enepa-II/public/action/eis/details?eisId=' + id, function( err, res, html ) {
+  Promise.map( eisIdArray, function( id ) {
```

Comment:
To chain promises for a queue of tasks, you could use this simple implementation: https://github.com/edgi-govdata-archiving/epa-geo-portal-database-scraper/blob/master/lib/enqueue.js

The usage would be:

```
function archive(id) {
  return request(...)
    .then(...); 
}

enqueue(eisIdArray, archive)
  .then(function () {
    console.log('All done!');
  );
```